zipline/_protocol.pyx

```diff
@@ -460,15 +462,25 @@ cdef class BarData:
             })
 
     cdef bool _can_trade_for_asset(self, asset, dt, adjusted_dt, data_portal):
-        if asset._is_alive(dt, False):
-            # is there a last price?
-            return not np.isnan(
-                data_portal.get_spot_value(
-                    asset, "price", adjusted_dt, self.data_frequency
-                )
-            )
+        session_label = normalize_date(dt) # FIXME
+        if not asset._is_alive_for_session(session_label):
```

Comment:
Naming-wise, looks like these underscore methods of `Asset` are actually public, in that consumers outside `Asset` call them?